.markdownlint-cli2.jsonc

```diff
@@ -0,0 +1,16 @@
+// https://github.com/DavidAnson/markdownlint-cli2#markdownlint-cli2jsonc
+{
+  "config": {
+    // https://github.com/DavidAnson/markdownlint/blob/main/README.md#rules--aliases
+    "MD013": false,
+    "MD024": false,
+    "MD033": false,
+    "MD041": false,
+    // XXX Rules to discuss:
+    "MD012": false // no-multiple-blanks, common in CHANGELOG.md files, ~1700 hits in changelog files
+    // "MD004": { "style": "dash" } // ul-style, ~3800 hits in 63 changelog files, XXX discuss just using "consistent" new default for dash
```

Comment:
@blumamir I'd like to discuss these two rules.

## `"MD012": false`

This is not disabled in the core repo. If re-enabled in the contrib repo here, there are approximately 1700 hits (a) in most of the 63 CHANGELOG.md files and (b) a few hits in these files:

```
   1 plugins/node/opentelemetry-instrumentation-mysql/README.md
   3 .github/ISSUE_TEMPLATE/plugin_request.md
   4 .github/ISSUE_TEMPLATE/feature_request.md
   6 .github/ISSUE_TEMPLATE/bug_report.md
```

I think the cases in the CHANGELOG.md files will just keep happening because they are generated by release-please, we do not want this rule enabled for them. There are two options:

1. globally disable the rule in this config file
2. locally disable the rule for each CHANGELOG.md file. This is slightly burdensome (just a onetime edit) because there are 63 changelog files.

Thoughts?

## `MD004`

Earlier we changed to `style: "dash"` to prefer dashes for bullets. In the core repo we disabled this rule locally in the *2* changelog files. Options:

1. Disable is locally in the 63 changelog files -- similar to how we did in the core repo.
2. Use the newer default of `"consistent"`, which ensures that each file is consistent in the bullets it uses. That means the rule could apply to CHANGELOG.md files as well, but would enforce consistent `*` usage. 
Thoughts?

Reply:
> I think the cases in the CHANGELOG.md files will just keep happening because they are generated by release-please, we do not want this rule enabled for them. There are two options:
> 
> 1. globally disable the rule in this config file
> 2. locally disable the rule for each CHANGELOG.md file. This is slightly burdensome (just a onetime edit) because there are 63 changelog files.
> 
> Thoughts?

Another alternative would be to use 2 markdown configs, one for CHANGELOG and one for other markdowns. I don't mind disabling it across the repo, but could be nice to have it if not too complex.



> Earlier we changed to `style: "dash"` to prefer dashes for bullets. In the core repo we disabled this rule locally in the _2_ changelog files. Options:

If we have different config file for CHANGELOGs than this is addressed as well, but if not straight-forward than we can disable it locally in each file or use the `"consistent"` option. What ever is easier and makes more sense to you

Reply:
I'd prefer to not have two separate config files, because:
- it would be more likely for those config files to diverge,
- it is a little messy to have yet another config file at the top,
- it will mean markdownlint editor plugins (like the VSCode one referred to above) won't work properly -- the plugin will just use the default `.markdownlint-cli2.jsonc` file

In commit 2e03f596 I've opted to avoid `<!-- markdownlist-disable ... -->` directives in every CHANGELOG.md file by selecting options that work for all .md files. That means `ul-style: consistent` and `no-multiple-blanks: false`.

(A personal preference note: I *like* using multiple blank lines sometimes, both in Markdown and code, to separate bigger sections vs. a single line for related blocks. So I was never a fan of `no-multiple-blank` lines and the eslint equivalent in code.)